tests/sota_tools/test-garage-deploy-missing-credentials

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -eu
+TARGET="Unable to read a-non-existent-file as archive or json file"
+# File not found and invalid/corrupt files are treated the same
+$1 --commit 16ef2f2629dc9263fdf3c0f032563a2d757623bbc11cf99df25c3c3f258dccbe -f a-non-existent-file -p a-non-existent-file --name testname -h hwids 2>&1 | grep "$TARGET"
```

Comment:
I think for this to be a valid test, you actually need to test the fetch and push credentials independently. It can still be done in one script if you run it twice and check both outputs.